nomt/src/page_cache.rs

```diff
@@ -435,6 +439,12 @@ impl PageCache {
     /// If the page is already in the cache, this method does nothing. Otherwise, it fetches the
     /// page from the underlying store and caches it.
     pub fn prepopulate(&self, page_id: PageId) {
+        if let Metrics::Active(metrics) = &self.shared.metrics {
```

Comment:
for a follow-up: I think these `match`es should be encapsulated inside the metrics type itself, with methods for bumping counts, creating timer guards, etc.

Reply:
Yes, I tried to pursue this, but I couldn't do it easily. That's because currently, from that match, we're able to directly access the Atomic types. If we wrap them inside any other type, such as a HashMap to simplify the access to metrics, we end up with a Send/Sync problem.

I have an idea of using enum + some glue code inside of metrics! I will push it soon!